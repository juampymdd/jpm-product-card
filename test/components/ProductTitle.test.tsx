
import React from 'React'

import renderer from 'react-test-renderer'
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/product';

describe('ProductTitle', () => {
    test('debe demostrar el componente correctamente con el tityulo personalizado', ()=>{
        const wrapper = renderer.create(
            <ProductTitle title="Custom Product"/>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('debe de mostrar el componente con el nombre del producto', () => {
        const wrapper = renderer.create(
            <ProductCard product={ product1 }>
                {
                    ()=>(
                    <ProductTitle title="Hola mundo"/>
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    })
    
})
