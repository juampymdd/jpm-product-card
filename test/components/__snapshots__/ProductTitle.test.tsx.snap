// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Hola mundo
  </span>
</div>
`;

exports[`ProductTitle debe demostrar el componente correctamente con el tityulo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;
